pilot/pkg/xds/debug.go

```diff
@@ -987,10 +987,30 @@ func (s *DiscoveryServer) forceDisconnect(w http.ResponseWriter, req *http.Reque
 	_, _ = w.Write([]byte("OK"))
 }
 
+func cloneProxy(proxy *model.Proxy) *model.Proxy {
+	if proxy == nil {
+		return nil
+	}
+
+	proxy.Lock()
+	defer proxy.Unlock()
+
+	out := &(*proxy)
```

Comment:
IIRC this doesn't actually perform a shallow copy, I guess it was optimized by the compiler

Reply:
Where did you see that

Reply:
You can run a simple test:
```go
type a struct {
	num int
}

func main() {
	x := &a{1}
	y := &(*x)
	println(x.num)	// 1
	y.num = 2
	println(x.num)	// 2
}
```
but if you change it to
```go
func main() {
	x := &a{1}
	y := *x
	z := &y
	println(x.num) // 1
	z.num = 2
	println(x.num) // 1
}
```
the output will be different

Reply:
Thanks for this important finding